File: LEETCODE/1965-sum-of-digits-in-base-k/sum-of-digits-in-base-k.cpp
class Solution {
public:
    int sumBase(int n, int k) {
        int s = 0;
        while(n > 0){
            s += n % k;
            n /= k;
        }
        return s;
    }
};


/*
class Solution {
public:
    int sumBase(int n, int k) {
        int s = 0;    // Counter variable to track sum
        while(n > 0){    // As long as the number is greater than 0:
            s += n % k;    // Add the remainder of n and k to the sum
            n /= k;    // Remove the last digit of n in base-k and store it in n
        }
        return s;    // Return the final sum
    }
};
*/

File: LEETCODE/1965-sum-of-digits-in-base-k/sum-of-digits-in-base-k.java
class Solution {
    public int sumBase(int n, int k) {
        int sum = 0;
        while(n > 0){
            int remainder = n % k;
            sum += remainder;
            n /= k;
        }
        return sum;
    }
}


/*
class Solution {
    public int sumBase(int n, int k) {
        int sum = 0;    // Variable to store the sum
        while(n > 0){    // As long as the number is greater than zero:
            int remainder = n % k;    // Get the remainder of n and k
            sum += remainder;    // Add it to the sum
            n /= k;    // Move to the next digit
        }
        return sum;    // Return the final sum of the digits
    }
}
*/